### PR TITLE
console: securityContext: add 'readOnlyRootFilesystem'

### PR DESCRIPTION
**Pull request by alfonsomthd**
Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=2166417

NGINX requires some dirs to be writable so we enable 2 `emptyDir` volumes for that purpose and update the NGINX config with the path of those dirs.

NOTE: when using the downstream image, we noticed this in the pod logs:
```/opt/app-root/etc/generate_container_user: line 6: /opt/app-root/etc/passwd: Read-only file system```
After a conversation with @SanjalKatiyar and Boris Ranto we decided to not take any action on this as it has no side effects on the behavior of the container.

### CODE REVIEW

File: console/nginx_conf.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ var NginxConf = `
 
 worker_processes auto;
 error_log /var/log/nginx/error.log;
-pid /run/nginx.pid;
+pid /var/lib/nginx/tmp/nginx.pid;
```
Comment (SanjalKatiyar) [inline console/nginx_conf.go:25]:
```suggestion
pid /var/lib/nginx/run/nginx.pid;
```
can't we keep naming same as original ?

Comment (SanjalKatiyar) [inline console/nginx_conf.go:25]:
```suggestion
pid /var/log/nginx/nginx.pid;
```
what if we add it to the same location as error.log ?
that way we don;t have to provide write permission for 2 separate paths...

Comment (SanjalKatiyar) [reply console/nginx_conf.go:25]: okay... if it needs access to this path anyway then we are good...

**Comment by SanjalKatiyar** [comment]
/retitle [WIP] console: securityContext: add 'readOnlyRootFilesystem'

**Comment by SanjalKatiyar** [comment]
/lgtm
cc @iamniting (no need to backport)

**Review by iamniting — APPROVED** [review]

**Pull request merged**